### PR TITLE
TC-MiguelRojas-02-08

### PR DESCRIPTION
**Pull request by airmiky**
Commits de Test case automatizados para:

18 TC-MiguelRojas-03 Update role test case automation
30 TC-MiguelRojas-04 Sickness list test case automation
27 TC-MiguelRojas-05 Create Sickness test case automation
28 TC-MiguelRojas-06 Sickness update test case automation
19 TC-MiguelRojas-07 Delete role test case automation
29 TC-MiguelRojas-08 Delete sickness test case automation

OBS: este commit esta en otro pull request con las observaciones resueltas, debido a que este branch se saco a partir del branch "TC-17", sus commits tambien estan presentes en este branch, el cual seria el "TC-MiguelRojas-02".

TC-17 Create role test case automation and add resource manager for SSIA

### CODE REVIEW

**Comment by airmiky** [comment]
18 TC-MiguelRojas-03 Update role test case automation

imagen

imagen

**Comment by airmiky** [comment]
30 TC-MiguelRojas-04 Sickness list test case automation

imagen

imagen

**Comment by airmiky** [comment]
27 TC-MiguelRojas-05 Create Sickness test case automation

imagen

imagen

**Comment by airmiky** [comment]
28 TC-MiguelRojas-06 Sickness update test case automation

imagen

imagen

**Comment by airmiky** [comment]
19 TC-MiguelRojas-07 Delete role test case automation

imagen

imagen

**Comment by airmiky** [comment]
29 TC-MiguelRojas-08 Delete sickness test case automation

imagen

imagen

File: src/main/java/org/umssdiplo/automationv01/core/managepage/role/RoleList.java [hunk 1]
```diff
@@ -13,6 +13,9 @@
     @FindBy(id = "newRoleBtn")
     private WebElement newRoleButton;
 
+    @FindBy(xpath = "//table[@id='roleTable']/descendant::tr[last()]/descendant::button[contains(concat(' ', normalize-space(@class), ' '), ' editBtn ')]")
+    private WebElement editButton;
+
     @FindBy(xpath = "//table[@id='roleTable']/descendant::tr[last()]/td[count(//table[@id='roleTable']/descendant::th[text()='Name'])]")
     private WebElement lastRoleName;
```
Comment (neyber) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/role/RoleList.java]: constructor method is not present.

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java [hunk 2]
```diff
@@ -87,6 +89,26 @@ public void createdRoleIsShowedInRoleListPage() throws Throwable {
         Assert.assertEquals(roleList.getLastRoleNameInTable(), DataDriverTest.readValues.getValue("Role.create.name"), "Fail, Role is not created");
     }
 
+    @And("^click in button 'Edit' of 'Role list' page$")
+    public void clickInButtonEditOfRoleListPage() throws Throwable {
```
Comment (neyber) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java]: Don't use 'In'

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java [hunk 3]
```diff
@@ -87,6 +89,26 @@ public void createdRoleIsShowedInRoleListPage() throws Throwable {
...
+        roleUpdate = roleList.clickEditButton();
+    }
+
+    @And("^update 'Role' in form using Data Driver Test on update 'Role' page$")
+    public void updateRoleInFormUsingDataDriverTestOnUpdateRolePage() throws Throwable {
+        roleUpdate.updateRoleUsingDataDriverTest();
+    }
+
+    @And("^click in button 'Update' into update 'Role' form page$")
```
Comment (neyber) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java]: don't use In

File: src/test/resources/features/role/RoleUpdate.feature [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# feature to review behaviour of update role
+Feature: RoleUpdate
+  Scenario: Roles - Actualizar datos de un rol
+    Given 'SSI-A' home page is loaded
+    And Clicking on Personnel menu on 'Header' page
+    And click in sub menu 'Roles' of menu 'personnel'
```
Comment (neyber) [inline src/test/resources/features/role/RoleUpdate.feature]: don't use in

File: src/test/resources/features/role/RoleUpdate.feature [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+    And click in button 'Edit' of 'Role list' page
+    And update 'Role' in form using Data Driver Test on update 'Role' page
+    And click in button 'Update' into update 'Role' form page
```
Comment (neyber) [inline src/test/resources/features/role/RoleUpdate.feature]: don't use in

**Review by neyber — COMMENTED** [review]
Please review observations.

File: src/main/java/org/umssdiplo/automationv01/core/managepage/menuheader/safetyMenu/SafetyMenu.java [hunk 6]
```diff
@@ -12,9 +13,17 @@
 */
 public class SafetyMenu extends BasePage {
```
Comment (neyber) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/menuheader/safetyMenu/SafetyMenu.java:15]: Add construct method.

File: src/main/java/org/umssdiplo/automationv01/core/managepage/sickness/SicknessList.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+ */
+package org.umssdiplo.automationv01.core.managepage.sickness;
+
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.umssdiplo.automationv01.core.managepage.BasePage;
+import org.umssdiplo.automationv01.core.utils.CommonEvents;
+
+public class SicknessList extends BasePage {
+
```
Comment (neyber) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/sickness/SicknessList.java:12]: Add construct method.

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java [hunk 8]
```diff
@@ -109,6 +110,19 @@ public void updatedRoleIsShowedInRoleListPage() throws Throwable {
         Assert.assertEquals(roleList.getLastRoleNameInTable(), DataDriverTest.readValues.getValue("Role.update.name"), "Fail, Role is not updated");
     }
 
+    //Sickness
+    private SicknessList sicknessList;
+
+    @And("^click in sub menu 'Sickness' of menu 'Safety'$")
```
Comment (neyber) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java]: don't use 'In'

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java [hunk 9]
```diff
@@ -109,6 +110,19 @@ public void updatedRoleIsShowedInRoleListPage() throws Throwable {
...
+    public void clickInSubMenuSicknessOfMenuSafety() throws Throwable {
```
Comment (neyber) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java]: don't use 'In'

File: src/test/resources/features/sickness/SicknessList.feature [hunk 10]
```diff
@@ -0,0 +1,7 @@
+# feature to review behaviour of sickness list
+Feature: SicknessList
+  Scenario: Enfermedades - Todos los datos de enfermedades son mostrados en una lista
+    Given 'SSI-A' home page is loaded
+    And click Safety 'menu' on 'Header' page
+    And click in sub menu 'Sickness' of menu 'Safety'
```
Comment (neyber) [inline src/test/resources/features/sickness/SicknessList.feature]: don't use 'In'

**Review by neyber — COMMENTED** [review]
Please review observations

File: src/main/java/org/umssdiplo/automationv01/core/managepage/sickness/SicknessList.java [hunk 11]
```diff
@@ -13,7 +13,19 @@
     @FindBy(id = "newSicknessBtn")
     private WebElement newSicknessButton;
 
+    @FindBy(xpath = "//table[@id='sicknessTable']/descendant::tr[last()]/td[count(//table[@id='sicknessTable']/descendant::th[text()='description']/preceding-sibling::th)+1]")
+    private WebElement lastSicknessDescription;
+
```
Comment (neyber) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/sickness/SicknessList.java]: add construct method.

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java [hunk 12]
```diff
@@ -123,6 +125,26 @@ public void sicknessListIsShowedInPage() throws Throwable {
         Assert.assertTrue(sicknessList.isSicknessListPresent(), "Fail, Sickness list is not loaded");
     }
 
+    @And("^click in button 'New Sickness' of sickness list page$")
```
Comment (neyber) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java]: don't use 'In'

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java [hunk 13]
```diff
@@ -123,6 +125,26 @@ public void sicknessListIsShowedInPage() throws Throwable {
...
+    public void clickInButtonNewSicknessOfSicknessListPage() throws Throwable {
+        sicknessCreate = sicknessList.clickNewSicknessButton();
+    }
+
+    @And("^fill 'Sickness' form using Data Driver Test on create 'Sickness' page$")
+    public void fillSicknessFormUsingDataDriverTestOnCreateSicknessPage() throws Throwable {
+        sicknessCreate.fillSicknessUsingDataDriverTest();
+    }
+
+    @And("^click in button 'Save' into 'Sickness' create form page$")
```
Comment (neyber) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java]: don't use 'In'

File: src/test/resources/features/sickness/SicknessCreate.feature [hunk 14]
```diff
@@ -0,0 +1,11 @@
+# feature to review behaviour of create sickness
+Feature: SicknessCreate
+  Scenario: Enfermedades - Registrar enfermedades para los empleados, y que se listen en la tabla
+    Given 'SSI-A' home page is loaded
+    And click Safety 'menu' on 'Header' page
+    And click in sub menu 'Sickness' of menu 'Safety'
```
Comment (neyber) [inline src/test/resources/features/sickness/SicknessCreate.feature]: don't use 'In'

File: src/test/resources/features/sickness/SicknessCreate.feature [hunk 15]
```diff
@@ -0,0 +1,11 @@
...
+    And click in button 'New Sickness' of sickness list page
```
Comment (neyber) [inline src/test/resources/features/sickness/SicknessCreate.feature]: don't use 'In'

File: src/test/resources/features/sickness/SicknessCreate.feature [hunk 16]
```diff
@@ -0,0 +1,11 @@
...
+    And fill 'Sickness' form using Data Driver Test on create 'Sickness' page
+    And click in button 'Save' into 'Sickness' create form page
```
Comment (neyber) [inline src/test/resources/features/sickness/SicknessCreate.feature]: don't use 'In'

**Review by neyber — COMMENTED** [review]
please review observations.

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java [hunk 17]
```diff
@@ -145,6 +147,26 @@ public void createdSicknessIsShowedInSicknessListPage() throws Throwable {
         Assert.assertEquals(sicknessList.getLastSicknessDescriptionInTable(), DataDriverTest.readValues.getValue("Sickness.create.description"), "Fail, Sickness is not created");
     }
 
+    @And("^click in button 'Edit' of 'Sickness list' page$")
+    public void clickInButtonEditOfSicknessListPage() throws Throwable {
```
Comment (neyber) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java]: don't use 'In'

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java [hunk 18]
```diff
@@ -145,6 +147,26 @@ public void createdSicknessIsShowedInSicknessListPage() throws Throwable {
...
+        sicknessUpdate = sicknessList.clickEditButton();
+    }
+
+    @And("^update 'Sickness' in form using Data Driver Test on update 'Sickness' page$")
+    public void updateSicknessInFormUsingDataDriverTestOnUpdateSicknessPage() throws Throwable {
+        sicknessUpdate.updateSicknessUsingDataDriverTest();
+    }
+
+    @And("^click in button 'Update' into update 'Sickness' form page$")
```
Comment (neyber) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java]: don't use 'In'

File: src/test/resources/features/sickness/SicknessUpdate.feature [hunk 19]
```diff
@@ -0,0 +1,11 @@
+# feature to review behaviour of update sickness
+Feature: SicknessUpdate
+  Scenario: Enfermedades - Actualizar enfermedades de los empleados y que se listen en la tabla
+    Given 'SSI-A' home page is loaded
+    And click Safety 'menu' on 'Header' page
+    And click in sub menu 'Sickness' of menu 'Safety'
```
Comment (neyber) [inline src/test/resources/features/sickness/SicknessUpdate.feature]: don't use 'In'

File: src/test/resources/features/sickness/SicknessUpdate.feature [hunk 20]
```diff
@@ -0,0 +1,11 @@
...
+    And click in button 'Edit' of 'Sickness list' page
+    And update 'Sickness' in form using Data Driver Test on update 'Sickness' page
+    And click in button 'Update' into update 'Sickness' form page
```
Comment (neyber) [inline src/test/resources/features/sickness/SicknessUpdate.feature]: don't use 'In'

**Review by neyber — COMMENTED** [review]
please review observations.

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java [hunk 21]
```diff
@@ -112,6 +114,21 @@ public void updatedRoleIsShowedInRoleListPage() throws Throwable {
         Assert.assertEquals(roleList.getLastRoleNameInTable(), DataDriverTest.readValues.getValue("Role.update.name"), "Fail, Role is not updated");
     }
 
+    @And("^click in button 'Delete' of 'Role list' page$")
```
Comment (neyber) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java]: don't use 'In'

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java [hunk 22]
```diff
@@ -112,6 +114,21 @@ public void updatedRoleIsShowedInRoleListPage() throws Throwable {
...
+    public void clickInButtonDeleteOfRoleListPage() throws Throwable {
+        roleDeleteAlert = roleList.clickDeleteButton();
+    }
+
+    @And("^click in button 'Accept' of delete 'Role' confirmation popup$")
```
Comment (neyber) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java]: don't use 'In'

File: src/test/resources/features/role/RoleDelete.feature [hunk 23]
```diff
@@ -0,0 +1,9 @@
+# feature to review behaviour of update role
+Feature: RoleDelete
+  Scenario: Roles - Eliminar datos de un rol
+    Given 'SSI-A' home page is loaded
+    And Clicking on Personnel menu on 'Header' page
+    And click in sub menu 'Roles' of menu 'personnel'
```
Comment (neyber) [inline src/test/resources/features/role/RoleDelete.feature]: don't use 'In'

File: src/test/resources/features/role/RoleDelete.feature [hunk 24]
```diff
@@ -0,0 +1,9 @@
...
+    And click in button 'Delete' of 'Role list' page
+    And click in button 'Accept' of delete 'Role' confirmation popup
```
Comment (neyber) [inline src/test/resources/features/role/RoleDelete.feature]: don't use 'In'

**Review by neyber — COMMENTED** [review]
please review observations.

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java [hunk 25]
```diff
@@ -184,6 +186,21 @@ public void updatedSicknessIsShowedInSicknessListPage() throws Throwable {
         Assert.assertEquals(sicknessList.getLastSicknessDescriptionInTable(), DataDriverTest.readValues.getValue("Sickness.update.description"), "Fail, Sickness is not updated");
     }
 
+    @And("^click in button 'Delete' of 'Sickness list' page$")
+    public void clickInButtonDeleteOfSicknessListPage() throws Throwable {
```
Comment (neyber) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java]: don't use 'In'

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java [hunk 26]
```diff
@@ -184,6 +186,21 @@ public void updatedSicknessIsShowedInSicknessListPage() throws Throwable {
...
+        sicknessDeleteAlert = sicknessList.clickDeleteButton();
+    }
+
+    @And("^click in button 'Accept' of delete 'Sickness' confirmation popup$")
+    public void clickInButtonAcceptOfDeleteSicknessConfirmationPopup() throws Throwable {
```
Comment (neyber) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java]: don't use 'In'

File: src/test/resources/features/sickness/SicknessDelete.feature [hunk 27]
```diff
@@ -0,0 +1,10 @@
+# feature to review behaviour of delete sickness
+Feature: SicknessDelete
+  Scenario: Enfermedades - Eliminar enfermedades de los empleados y que dejen de listarse en la tabla
+    Given 'SSI-A' home page is loaded
+    And click Safety 'menu' on 'Header' page
+    And click in sub menu 'Sickness' of menu 'Safety'
```
Comment (neyber) [inline src/test/resources/features/sickness/SicknessDelete.feature]: don't use 'In'

File: src/test/resources/features/sickness/SicknessDelete.feature [hunk 28]
```diff
@@ -0,0 +1,10 @@
...
+    And click in button 'Delete' of 'Sickness list' page
+    And click in button 'Accept' of delete 'Sickness' confirmation popup
```
Comment (neyber) [inline src/test/resources/features/sickness/SicknessDelete.feature]: don't use 'In'

**Review by neyber — COMMENTED** [review]
please review observations.

File: src/main/java/org/umssdiplo/automationv01/core/managepage/menuheader/safetyMenu/SafetyMenu.java (hunk 6 above)
Comment (airmiky) [reply src/main/java/org/umssdiplo/automationv01/core/managepage/menuheader/safetyMenu/SafetyMenu.java:15]: El owner de SafetyMenu es henry, el hara el fix

**Comment by airmiky** [comment]
Neyber.. se ha hecho fix de las observaciones.

**Pull request merged**